plugins/inputs/intel_powerstat/README.md

```diff
@@ -0,0 +1,199 @@
+# Intel PowerStat Input Plugin
+
+Telemetry frameworks allow users to monitor critical platform level metrics. 
+Key source of platform telemetry is power domain that is beneficial for MANO/Monitoring&Analytics systems 
+to take preventive/corrective actions based on platform busyness, CPU temperature, actual CPU utilization 
+and power statistics. Main use cases are power saving and workload migration.
+
+Intel PowerStat plugin supports Intel based platforms and assumes presence of Linux based OS.
+
+### Configuration:
+```toml
+# Intel PowerStat plugin enables monitoring of platform metrics (power, TDP) and per-CPU metrics like temperature, power and utilization.
+[[inputs.intel_powerstat]]
+  ## All global metrics are always collected by Intel PowerStat plugin.
+  ## User can choose which per-CPU metrics are monitored by the plugin in cpu_metrics array.
+  ## Empty array means no per-CPU specific metrics will be collected by the plugin - in this case only platform level
+  ## telemetry will be exposed by Intel PowerStat plugin.
+  ## Supported options:
+  ## "cpu_frequency", "cpu_c1_state_residency", "cpu_c6_state_residency", "cpu_busy_cycles", "cpu_temperature", "cpu_busy_frequency"
+  # cpu_metrics = []
+```
+### Example: Configuration with no per-CPU telemetry
+This configuration allows getting global metrics (processor package specific), no per-CPU metrics are collected:
+```toml
+[[inputs.intel_powerstat]]
+  cpu_metrics = []
+```
+
+### Example: Configuration for CPU Temperature and Frequency only
+This configuration allows getting global metrics plus subset of per-CPU metrics (CPU Temperature and Current Frequency):
+```toml
+[[inputs.intel_powerstat]]
+  cpu_metrics = ["cpu_frequency", "cpu_temperature"]
+```
+
+### Example: Configuration with all available metrics
+This configuration allows getting global metrics and all per-CPU metrics:
+```toml
+[[inputs.intel_powerstat]]
+  cpu_metrics = ["cpu_frequency", "cpu_c1_state_residency", "cpu_c6_state_residency", "cpu_busy_cycles", "cpu_temperature", "cpu_busy_frequency"]
+```
+
+### SW Dependencies:
+Plugin is based on Linux Kernel modules that expose specific metrics over `sysfs` or `devfs`.
+The following dependencies are expected by plugin:
+- _intel-rapl_ module which exposes Intel Runtime Power Limiting metrics over `sysfs` (`/sys/devices/virtual/powercap/intel-rapl`)
+Minimum kernel version required is 3.13 (to make sure RAPL driver is included in current distro),
+- _msr_ kernel module that provides access to processor model specific registers from user space, 
```

Comment:
Can you also provide a vanilla kernel version here?

Reply:
It's really hard to find exact version :D For sure _msr_ kernel module worked for 2.6 (~15 years ago), maybe even for earlier versions.
But you need at least 3.1 to run Go application for Linux (and this plugin aims only for Linux): https://github.com/golang/go/wiki/MinimumRequirements
So I would say: "If you can run Telegraf on Linux, you should have access to _msr_ kernel module" and I would not write anything here in README.

Reply:
Would be fine if you put "3.1 maybe earlier". People < 3.1 then know it might or might not work. It's about the "no worries it will work anyway" portion of ppl. It's just you put a version on any other module... I'd like to see the "first-to-be-sure" version here if it exists.

Reply:
Section restructured, "first-to-be-sure" version added (globally, for types of metrics)